ironfish/src/utils/bloomFilter.ts

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+function isPowerOf2(n: number): boolean {
+  // This checks that n is non-zero, and that n and n-1 have no bits in common
+  // (taken from https://stackoverflow.com/a/30924333)
+  return !!(n && !(n & (n - 1)))
+}
+
+export class BloomFilter {
```

Comment:
Forgot to add a doc comment here explaining the design choices & the assumptions. I might also add a `hashFn` parameter to the constructor that equals the identity function by default.